pkg/config/config.go

```diff
@@ -34,19 +37,87 @@ type IgnoreEntry struct {
 	Reason      string    `toml:"reason"`
 }
 
+type IgnorePackageVersionEntry struct {
+	Name        string    `toml:"name"`
+	Version     string    `toml:"version"`
+	Ecosystem   string    `toml:"ecosystem"`
+	IgnoreUntil time.Time `toml:"ignoreUntil"`
+	Reason      string    `toml:"reason"`
+}
+
+type OverridePackageVersionLicenseEntry struct {
+	Name            string   `toml:"name"`
+	Major           string   `toml:"major"`
+	Minor           string   `toml:"minor"`
+	Patch           string   `toml:"patch"`
+	ExactVersion    string   `toml:"exactVersion"`
+	Ecosystem       string   `toml:"ecosystem"`
+	LicenseOverride []string `toml:"licenseOverride"`
+	Reason          string   `toml:"reason"`
+}
+
 func (c *Config) ShouldIgnore(vulnID string) (bool, IgnoreEntry) {
 	index := slices.IndexFunc(c.IgnoredVulns, func(elem IgnoreEntry) bool { return elem.ID == vulnID })
 	if index == -1 {
 		return false, IgnoreEntry{}
 	}
 	ignoredLine := c.IgnoredVulns[index]
-	if ignoredLine.IgnoreUntil.IsZero() {
+
+	return shouldIgnoreTimestamp(ignoredLine.IgnoreUntil), ignoredLine
+}
+
+func (c *Config) ShouldIgnorePackageVersion(name, version, ecosystem string) (bool, IgnorePackageVersionEntry) {
+	index := slices.IndexFunc(c.IgnoredPackageVersions, func(elem IgnorePackageVersionEntry) bool {
+		if name != elem.Name {
+			return false
+		}
+
+		return version == elem.Version || elem.Version == ""
+	})
+	if index == -1 {
+		return false, IgnorePackageVersionEntry{}
+	}
+	ignoredLine := c.IgnoredPackageVersions[index]
+
+	return shouldIgnoreTimestamp(ignoredLine.IgnoreUntil), ignoredLine
+}
+
+func (c *Config) ShouldOverridePackageVersionLicense(name, version, ecosystem string) (bool, OverridePackageVersionLicenseEntry) {
+	versionParts := strings.Split(version, ".")
```

Comment:
Not all ecosystems follow semver, not sure how well this will handle non semver versions. Given we have an ecosystem string, can you use the semantic library we have to parse the versions here?

Reply:
Gareth might have some ideas here as well.

Reply:
Ah yes we can, I flagged this as something I would come back to this week: https://github.com/google/osv-scanner/issues/814#issuecomment-2078688416

Reply:
Since it's already a dependency, let's use the deps.dev semver package instead of having to import another package.

Reply:
Ah sorry I misread - you suggested the `semantic` library. @G-Rath suggested this too in the thread https://github.com/google/osv-scanner/issues/814#issuecomment-2078744502

It supports way more ecosystems than deps.dev's semver library so it might be the better choice actually.

Reply:
@G-Rath what do you think is the best way to specify a version prefix range?

I'm thinking an array of [components](https://github.com/G-Rath/osv-detector/blob/e04029c23b143dadd16cdcf0c801773fec58f3d6/pkg/semantic/version.go#L15)?
so [3, 14] means 3.14.x.

But I want to hear other thoughts/suggestions.

Reply:
I'm not keen to start making a requirement language for each ecosystem, and I noticed that Components is already an abstraction in the `semantic` library. But I haven't read the library too in depth to figure out if there is a more natural way than specifying an array.

Reply:
Rewording what @another-rex suggested offline:

```
[[Package]]
name = "pkg-name"
[[Package.requirement]]
operator = ">"
version = "1.0.0"
[[Package.requirement]]
operator = "<"
version = "1.1.0"
```

since the `semantic` library has a `Compare` function.

Reply:
Pros:
> less fields in the config, more flexibility, easier to understand, and probably less implementation as well

Reply:
fwiw, I personally wonder if it might be too much to be including version ranges in at least this PR - I understand why they could be useful but they don't feel critical to the general feature while requiring you to deal with a bunch of extra complexity.

Having said that, I think your latest proposed config could be a good middle ground since it's hard to argue about supporting more complexities like "1.0.x" for the time being - and frankly while I have a few ideas on how you could do ranges, the more I think about it the more I feel your proposal is good by allowing you to side step a bunch of stuff like differences between version ecosystems.

(part of the reason I say this too is because we're using Go which I've found doesn't combine well with "multiple ecosystems" in this way as everything has to be a single type - that's overall fine, but just means you can't sweep little nounces between ecosystems under the rug for the time being like you could with say TypeScript; and then combine with this being a public cli... 😅)

Reply:
The range syntax makes sense to me, but +1 to @G-Rath's comment to skip this at least in this PR. Is there strong demand for such a feature? 

It adds a bit of complexity to both the config spec and our maintenance.

Reply:
SGTM. So no ranges, you can only specify an exact version (named `version`). If a version is omitted, all versions of the package match.